cwl/src/main/scala/cwl/CwlWomExpression.scala

```diff
@@ -122,14 +121,17 @@ object InitialWorkDirFileGeneratorExpression {
 
     implicit val caseExpression: Case.Aux[Expression, InitialWorkDirFileEvaluator] = {
       at { expression =>
-        (_, unmappedParameterContext, mappedInputValues, expressionLib) => {
+        (ioFunctionSet, unmappedParameterContext, _, expressionLib) => {
           // A single expression which must evaluate to an array of Files
           val expressionEvaluation = ExpressionEvaluator.eval(expression, unmappedParameterContext, expressionLib)
 
           expressionEvaluation flatMap {
             case array: WomArray if WomArrayType(WomSingleFileType).coercionDefined(array) =>
               WomArrayType(WomSingleFileType).coerceRawValue(array).toErrorOr
-            case file: WomFile => file.valid
+            case file: WomFile =>
+              // The file should be staged to the initial work dir using the base filename.
+              val baseFileName = file.value.substring(file.value.lastIndexOf('/') + 1)
```

Comment:
You presumably also need to do the same transform to the elements of the array in the `case` above this one?

Reply:
Maybe not for this conformance test though perhaps for my conscience.